### PR TITLE
Make GSPH package optional

### PR DESCRIPTION
**Pull request by adayton1**
# Summary

- This PR is a feature
- It does the following:
  - Makes the GSPH package optional to reduce build times and library size
  - Moves GSPH implementation of computeSPHVolume to Hydro
  - Moves GSPH implementation of ReplaceWithRatioPolicy to DataBase
  - Filters out tests that use GSPH if it is disabled
  - Fixes https://github.com/LLNL/spheral/issues/292

------
### ToDo :

- [x] Annotate ``RELEASE_NOTES.md`` with notable changes.
- [x] Create LLNLSpheral PR pointing at this branch. (PR# https://rzlc.llnl.gov/gitlab/owen/LLNLSpheral/-/merge_requests/99)
- [x] LLNLSpheral PR has passed all tests.

### CODE REVIEW

**Review by ptsuji — DISMISSED** [review]

**Comment by adayton1** [comment]
I moved files to locations that @jmpearl suggested. These ones do have to be moved since they are actually shared by multiple packages. But as far as I can tell, the files that are moved are internal and not part of the user-facing API.

**Review by ptsuji — DISMISSED** [review]

**Comment by adayton1** [comment]
Any comments?

**Comment by ptsuji** [comment]
I think you want a review from @jmikeowen and @jmpearl

**Comment by jmpearl** [comment]
> Any comments?

The edits look good to me, shoot me an email when you're read for the final review. I don't always get pinged by git when I have action items.

File: src/Hydro/computeSPHVolume.cc
Comment (jmpearl) [inline src/Hydro/computeSPHVolume.cc:1]: is #include "Field/Field.hh" needed in this one?

Comment (adayton1) [reply src/Hydro/computeSPHVolume.cc:1]: It is needed, but it is also brought in by FieldList.hh so I can remove it if you want.

Comment (adayton1) [reply src/Hydro/computeSPHVolume.cc:1]: I removed it.

**Review by jmpearl — APPROVED** [review]
This looks good to me

**Review by ptsuji — APPROVED** [review]

**Pull request merged**